data_model/src/sorting.rs

```diff
@@ -0,0 +1,68 @@
+//! Structures and traits related to sorting.
+
+#[cfg(not(feature = "std"))]
+use alloc::{
+    borrow::ToOwned as _,
+    collections::btree_map,
+    format,
+    string::{String, ToString as _},
+    vec,
+    vec::Vec,
+};
+#[cfg(feature = "std")]
+use std::collections::btree_map;
+
+use serde::{Deserialize, Serialize};
+#[cfg(feature = "warp")]
+use warp::{Filter, Rejection};
+
+use crate::prelude::*;
+
+const SORT_BY_KEY: &str = "sort_by_key";
+
+/// Structure for sorting requests
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct Sorting {
+    /// [`Name`] of the key in [`Asset`]'s metadata used to order query result.
+    pub sort_by_key: Option<Name>,
+}
+
+impl Sorting {
+    ///
+    pub fn new(key: Name) -> Self {
+        Self {
+            sort_by_key: Some(key),
+        }
+    }
+}
+
+impl From<Sorting> for btree_map::BTreeMap<String, String> {
+    fn from(sorting: Sorting) -> Self {
+        let mut btree = Self::new();
+        if let Some(key) = sorting.sort_by_key {
+            btree.insert(String::from(SORT_BY_KEY), key.to_string());
+        }
+        btree
+    }
+}
+
+impl From<Sorting> for Vec<(&'static str, String)> {
+    fn from(sorting: Sorting) -> Self {
+        let mut vec = Vec::new();
+        if let Some(key) = sorting.sort_by_key {
+            vec.push((SORT_BY_KEY, key.to_string()));
+        }
+        vec
+    }
+}
+
+#[cfg(feature = "warp")]
+/// Filter for warp which extracts sorting
+pub fn sorting() -> impl Filter<Extract = (Sorting,), Error = Rejection> + Copy {
```

Comment:
shouldn't we define this in torii?

Reply:
I think so but `Pagination` defined also in data_model so I decided to define `Sorting` here. It looks weird but in my opinion it should be refactored in another issue.

Reply:
These should be extracted into their own `query` crate: Pagination, Predicate, Sorting and the queries themselves.